### PR TITLE
Update dataset_utils.py

### PR DESCRIPTION
**Pull request by GeoffCan**
Many people have an issue with tf.keras.utils.image_dataset_from_directory not working when a label list is supplied rather than using a directory structure to define classes. Adding this simple change locally means that class_names contains the classes supplied in 'labels' in the initial call, rather than just the parent/top directory name, and solved the problem going forward. Hopefully it can be incorporated??

### CODE REVIEW

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! Please add a unit test to check the fix and to avoid breaking it in the future. Also, format the code via `sh shell/format.sh`.

**Comment by GeoffCan** [comment]
@fchollet I'm not from a software engineering background, but I sort of get what you asking. To clarify I need to fork and edit "image_dataset_utils_test.py" so that it can check for changes and also remove the comments from my original edit of "dataset_utils.py"??

**Comment by fchollet** [comment]
Part of why I'm asking for a test is I don't know what issue this fixes. 

> "tf.keras.utils.image_dataset_from_directory not working when a label list is supplied rather than using a directory structure to define classes. Adding this simple change locally means that class_names contains the classes supplied in 'labels' in the initial call, rather than just the parent/top directory name"

In this case, the `class_names` attribute is intended to be the sorted list of unique labels. Is that not the case?

**Comment by GeoffCan** [comment]
It isn't the case. :( It returns the sub-directory structure as class names and ignores the labels supplied. My little "tweak" solved that locally for me. The other tweak I had to make was to provide one sub-directory for "image_dataset_from_directory" to look into to find my images ie: Train ->train_img->image1.png, image2.png etc... Rather than just providing one directory ie: Train_img->img1.png, image2.png etc.... When I look at how the "test" directories are created I can't see how to replicate this situation, and whether that is a good way for the function to operate syntactically(??) also??? This has led me to look into the actual logic of the function and how it determines what to do based on what is supplied to the "labels" variable....

**Comment by GeoffCan** [comment]
Some further checks indicates that the directory structure isn't a problem...just how the labels are parsed

**Comment by GeoffCan** [comment]
I've made a minor change to image_dataset_utils_test.py to hopefully illustrate the issue, and created a pull request.

**Comment by fchollet** [comment]
Let's move the discussion to the other PR.
